win32/src/job_object_helper.c

```diff
@@ -11,7 +11,130 @@
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_005: [ If percent_cpu is greater than 100, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+    if ((percent_physical_memory == 0) ||
+        (percent_physical_memory > 100) || 
+        (percent_cpu == 0) ||
+        (percent_cpu > 100))
+    {
+        LogError("Invalid arguments:"
+            " uint32_t percent_physical_memory = %d"
+            ", uint32_t percent_cpu = %d",
+            percent_physical_memory,
+            percent_cpu);
+        /*Codes_SRS_JOB_OBJECT_HELPER_18_015: [ If there are any failures, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+        result = MU_FAILURE;
+    }
+    else
+    {
+        /*Codes_SRS_JOB_OBJECT_HELPER_18_006: [ job_object_helper_limit_resources shall call CreateJobObject to create a new job object passing NULL for both lpJobAttributes and lpName. ]*/
+        HANDLE job_object = CreateJobObject(NULL, NULL);
+        if (job_object == NULL)
+        {
+            LogLastError("CreateJobObject(NULL, NULL) failed");
+            /*Codes_SRS_JOB_OBJECT_HELPER_18_015: [ If there are any failures, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+            result = MU_FAILURE;
+        }
+        else
+        {
+            /*Codes_SRS_JOB_OBJECT_HELPER_18_007: [ job_object_helper_limit_resources shall call GetCurrentProcess to get the current process handle. ]*/
+            HANDLE current_process = GetCurrentProcess();
+            if (current_process == NULL)
+            {
+                LogLastError("GetCurrentProcess() failed");
+                /*Codes_SRS_JOB_OBJECT_HELPER_18_015: [ If there are any failures, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+                result = MU_FAILURE;
+            }
+            else
+            {
+                /*Codes_SRS_JOB_OBJECT_HELPER_18_008: [ job_object_helper_limit_resources shall call AssignProcessToJobObject to assign the current process to the new job object. ]*/
+                if (AssignProcessToJobObject(job_object, current_process) == FALSE)
+                {
+                    LogLastError("AssignProcessToJobObject(HANDLE job_object = %p, HANDLE current_process = %p) failed", job_object, current_process); 
+                    /*Codes_SRS_JOB_OBJECT_HELPER_18_015: [ If there are any failures, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+                    result = MU_FAILURE;
+                }
+                else
+                {
+                    MEMORYSTATUSEX memory_status_ex;
+                    memory_status_ex.dwLength = sizeof(memory_status_ex);
+                    /*Codes_SRS_JOB_OBJECT_HELPER_18_009: [ job_object_helper_limit_resources shall call GlobalMemoryStatusEx to get the total amount of physical memory. ]*/
+                    if (!GlobalMemoryStatusEx(&memory_status_ex))
```

Comment:
>GlobalMemoryStatusEx [](http://example.com/codeflow?start=25&length=20)

I would postpone line 61 until after this line.

Reply:
OK, I can see that. Easy enough.